### PR TITLE
Fix loading of all migrations

### LINKED ISSUE #89: data:rollback fails depending on order of data and schema migrations
I run into the following issue:
When the order of data and schema migrations (based on filename) is intermixed, I cannot do a data:rollback if there are schema migrations in between the current and previous data migration.

Consider the following output of `db:migrate:status:with_data`
 ```
Status    Type    Migration ID   Migration Name
------------------------------------------------------------
   up     schema  20160517153654  Create links
   up     schema  20160517153831  Create assets
   up     schema  20161202212757  Devise create users
   up      data   20169423231126  Add users
   up      data   20169517225124  Add basic data
   up     schema  20171108210818  Create factions
   up     schema  20171116173510  Create entries
   up      data   20180517231700  Add draw events
```
At this point, I can do a `data:rollback` to revert the latest data migration. However, any further calls to `data:rollback` will silently fail. The overall behavior and log messages are as described in #67 .

I would expect that `data:rollback` would not be affected by schema migration at all. Is this normal?

### PR DESCRIPTION
**Pull request by ilyakatz**
Fixes #89, #86

### CODE REVIEW

**Comment by andela-denoma** [comment]
So basically, this PR is getting only data migration versions so a rollback doesn't fail right?

**Comment by ilyakatz** [comment]
yes, a long time ago, it used to write data migrations into schema table. v2 stopped doing that, and now it's time to cut it off that support since it's causing other problems.

**Comment by ilyakatz** [comment]
@andela-denoma if you have time, can you play around with this and see if you notice anything suspicious. I think this is done but I'm a bit paranoid about releasing it 😨

File: tasks/databases.rake [hunk 1]
```diff
@@ -1,4 +1,5 @@
 require 'data_migrate/tasks/data_migrate_tasks'
+require 'pry'
```
Comment (ilyakatz) [inline tasks/databases.rake]: need to remove this

**Comment by andela-denoma** [comment]
I'm not exactly sure how to reproduce this issue. This were my testing steps

Test one
1. Create a data migration
2 Create a schema migration

rake data:rollback doesn't rollback any of the migrations

Test two
1. Create a data migration
2 Create a schema migration
3. Create a data migration

rake data:rollback rolls back the data migration

**Comment by ilyakatz** [comment]
For testing, 

1. I took a long running project (one that was around from before v2) @EnomaDebby, you know the one i'm talking about ;)

1. ran `rake data:migrate`. I started seeing old migrations from 2015

1. opened rails console  and ran `DataMigrate::LegacyMigrator.new.migrate`

1. ran `rake data:migrate`. Old migrations no longer show up.

**Pull request merged**